lucet-runtime/lucet-runtime-internals/src/module/dl.rs

```diff
@@ -117,6 +117,36 @@ impl DlModule {
             }
         };
 
+        // resolve imports (oh no ...)
```

Comment:
This is the really questionable change. Since looking up exported functions is done by walking `function_manifest`, where an imported function has address `0` (no relocation to place there, so it's just left default), we need to fix this up at module instantiation time with the right hostcall addresses.

The other option I can see is inserting stub functions that just `jmp [import_addr]` that we can actually export, which trades off instantiation time for the extra indirection at runtime. I'm just not sure how that tradeoff plays out. If we don't declare the import in artifact headers anymore, and fix up addresses here, we're just avoiding work the loader would have done, so the net instantiation time _should_ be the same.